components/collapsible/composed/src/index.js

```diff
@@ -0,0 +1,57 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+class CollapsibleComposed extends Component {
+  state = { collapsed: this.props.collapsed }
+
+  _toggle () {
+    const { collapsed } = this.state
+    this.setState({ collapsed: !collapsed })
+    this.props.onToggle(!collapsed)
+  }
+
+  render () {
+    const { hiddenItems, inline, items, label } = this.props
+    const { collapsed } = this.state
+    const collapseIcon = collapsed ? '+' : '-'
+
+    const itemClass = cx('sui-CollapsibleComposed-item', { 'sui-CollapsibleComposed-item--inline': inline })
+    const togglerClass = cx('sui-CollapsibleComposed-toggler', { 'sui-CollapsibleComposed-item--inline': inline })
+
+    const displayItems = (items || []).concat((!collapsed && hiddenItems) || []).map(item => {
```

Comment:
That's hardly readable. First, I would put `items` defaultProp as an empty array in order to avoid the first `||`.

```
const displayItems = (!collapsed && hiddenItems) || []
const allItems = items.concat(displayItems)
const displayItemsToRender = allItems.map(item => <div className={itemClass}>{item}</div>)
```

Variable naming is opinable but I would like to focus on the one responsability per line in order to improve the readability of the code.

Reply:
items is already set as [] in the default props. I'll split the line into several declarations to make it clear.